crates/router/src/connector/nmi/transformers.rs

```diff
@@ -525,13 +525,13 @@ impl TryFrom<types::PaymentsSyncResponseRouterData<types::Response>>
             status: enums::AttemptStatus::from(NmiStatus::from(response.transaction.condition)),
             response: Ok(types::PaymentsResponseData::TransactionResponse {
                 resource_id: types::ResponseId::ConnectorTransactionId(
-                    response.transaction.transaction_id,
+                    response.transaction.transaction_id.clone(),
                 ),
                 redirection_data: None,
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: item.response.orderid.or(Some(item.response.transactionid)),
```

Comment:
```suggestion
                connector_response_reference_id: None,
```

Reply:
Hey @TejasMate 
Thanks for the quick changes.
Can you also do the below change.

Also in line 348 of this file make it as(StandardResponse Struct) :
`pub orderid: Option<String>,`

Also run the following command before you commit the changes to handle the code formatting
`cargo +nightly fmt`